go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -887,6 +887,7 @@ func (h *defaultNetworkControllerEventHandler) UpdateResource(oldObj, newObj int
 					nodeGatewayMTUSupportChanged(oldNode, newNode))
 				_, hoSync := h.oc.hybridOverlayFailed.Load(newNode.Name)
 				_, syncZoneIC := h.oc.syncZoneICFailed.Load(newNode.Name)
+				syncZoneIC = syncZoneIC || util.NodeNetworkIDAnnotationChanged(oldNode, newNode, h.oc.GetNetworkName())
```

Comment:
Is this relevant here? It exists in the secondaryLayer3NetworkController code!

Reply:
In practice, this is constant for the default network, won't change and we are already waiting for it to be set on the main thread or bailing out, so `NodeNetworkIDAnnotationChanged` would never return true. But I guess it doesn't hurt to have it in case this changes in the future